pkg/controller/clusterconfig/clusterconfig_controller.go

```diff
@@ -108,6 +109,18 @@ func (r *ReconcileClusterConfig) Reconcile(ctx context.Context, request reconcil
 	network.MergeClusterConfig(&operConfig.Spec, clusterConfig.Spec)
 
 	if _, ok := clusterConfig.Annotations[names.NetworkTypeMigrationAnnotation]; ok {
+		// Only allows live migration on SD managed clusters
+		inManagedCluster, err := util.IsRunningInManagedCluster(ctx, r.client)
+		if err != nil {
+			log.Println(err)
+			return reconcile.Result{}, err
+		}
+		if !inManagedCluster {
+			err := fmt.Errorf("Network type live migration is not supported on self-managed clusters")
+			r.status.SetDegraded(statusmanager.ClusterConfig, "NetworkTypeMigrationFailed",
+				fmt.Sprintf("Failed to process network type live migration (%v).Use 'oc edit network.config.openshift.io cluster' to fix.", err))
+			return reconcile.Result{}, err
+		}
```

Comment:
So in theory, we could just do the check in the oper config controller and not here. I am not sure if I am missing why this additional check is being done here, unless it is just to give a more meaningful error? But then (next comment...)

Reply:
haha good question, I had the same thought but then on my side I just don't know exactly what the cluster-config oper does which operconfig doesn't.

Reply:
I dont know why its there as well - removing it as its not needed.